parflow_data_management/scheduler/models/cluster.py

```diff
@@ -5,17 +5,20 @@
 from django_extensions.db.models import TimeStampedModel
 from django.utils.translation import gettext_lazy as _
 from guardian.shortcuts import assign_perm
+from paramiko import RSAKey
 
+from transport.models.ssh_connection_info import SSHConnectionInfo
 
 class Cluster(TimeStampedModel, models.Model):
     name = CharField(_("Name of Cluster"), blank=True, max_length=255)
-    user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name="clusters")
-    # TODO: more fields
+    owner = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name="clusters")
+    # TODO: Keep this nullable?
+    ssh_info = models.OneToOneField(SSHConnectionInfo, null=True, on_delete=models.SET_NULL)
```

Comment:
So I think that a cluster is something that can either be private or share with a group of users. In the case that it is shared each user will use their own key to access it, so this wouldn't be a one-to-one relationship. Or may be the key cluster relationship should be held else where?

Reply:
Good point. What about something like a `credentials` model? Storing a `user` and `key_pair`, and possibly other fields. Then each cluster could have a list of `credentials`, corresponding to `users` that can access the cluster and their `key_pairs`.

Reply:
We basically want to model the authorized keys file. Indicating the keys that can access the cluster and therefore the users that have access.

Reply:
That would be pretty straightforward. Should this new model store `key_pairs` or just `public_keys`, like a true authorized keys file?

Reply:
I think just `public_keys`

Reply:
How does 8700a67d652a6a781cd84e20941ec69ad640535d look?